service/dap/server.go

```diff
@@ -1688,36 +1700,35 @@ func (s *Server) onStackTraceRequest(request *dap.StackTraceRequest) {
 		isSystemGoroutine = g.System(s.debugger.Target())
 	}
 
-	stackFrames := make([]dap.StackFrame, len(frames))
-	for i, frame := range frames {
+	var stackFrames []dap.StackFrame
```

Comment:
We had a null bug when variables array was no longer explicitly set to empty (fix at #2592). Can something like this happen here? I guess we don't accept 0 depth, but can we get an empty stack without an error? Might be safest to just initialize to empty just in case.